src/main/java/subway/member/domain/MemberFavorites.java

```diff
@@ -0,0 +1,62 @@
+package subway.member.domain;
+
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import subway.constant.SubwayMessage;
+import subway.exception.SubwayBadRequestException;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Embeddable;
+import javax.persistence.OneToMany;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Getter
+@Embeddable
+@NoArgsConstructor
+public class MemberFavorites {
+
+    @OneToMany(mappedBy = "member", cascade = CascadeType.PERSIST, orphanRemoval = true)
+    private final List<Favorite> favorites = new ArrayList<>();
+
+    public void add(Favorite newFavorite, Member member) {
+        validAlreadyExist(newFavorite);
+
+        newFavorite.setMember(member);
+        add(newFavorite);
+    }
+
+    public boolean IsExistFavorite(Favorite favorite) {
+        return favorites.contains(favorite);
+    }
+
+    public void removeFavorite(Member member, Favorite favorite) {
+        if (!member.IsExistFavorite(favorite)) {
+            throw new SubwayBadRequestException(SubwayMessage.FAVORITE_NOT_MY_OWN);
+        }
+        remove(favorite);
+    }
+
+    private void validAlreadyExist(Favorite newFavorite) {
+        Optional<Favorite> findAnyFavorite = findFavoriteBySourceAndTarget(newFavorite);
+        if (findAnyFavorite.isPresent()) {
+            throw new SubwayBadRequestException(SubwayMessage.FAVORITE_IS_ALREADY_EXIST);
+        }
+
+    }
+
+    private Optional<Favorite> findFavoriteBySourceAndTarget(Favorite newFavorite) {
+        return favorites.stream()
+                .filter(favorite -> favorite.getSourceStation().equals(newFavorite.getSourceStation()) || favorite.getTargetStation().equals(newFavorite.getTargetStation()))
+                .findAny();
```

Comment:
```suggestion
        return favorites.stream()
                .anyMatch(favorite -> favorite.getSourceStation().equals(newFavorite.getSourceStation()) || favorite.getTargetStation().equals(newFavorite.getTargetStation()));
```

로 바꾸면 조금 더 간결해질 것 같아요 😄